src/utils/time-utils.ts

```diff
@@ -0,0 +1,107 @@
+import {
+    CAPTURE_INTERVAL_RE,
+    DURATION_RE,
+    POSTGRES_INTERVAL_RE,
+} from 'validation';
+import { hasLength } from './misc-utils';
+
+interface SegmentedInterval {
+    h: number;
+    m: number;
+    s: number;
+}
+
+const parsePostgresInterval = (interval: string): SegmentedInterval => {
+    const [hours, minutes, seconds] = interval.split(':').map((segment) => {
+        const numericSegment = Number(segment);
+
+        return isFinite(numericSegment) ? numericSegment : -1;
+    });
+
+    return { h: hours, m: minutes, s: seconds };
+};
+
+const parseCaptureInterval = (interval: string): SegmentedInterval => {
+    let hours = -1;
+    let minutes = -1;
+    let seconds = -1;
+
+    interval.split(' ').forEach((segment) => {
+        const numericSegment = Number(segment.substring(0, segment.length - 1));
+        const numericValue = isFinite(numericSegment) ? numericSegment : -1;
```

Comment:
We repeat this check so should probably just throw into a function.

Reply:
We should also probably use `isSafeInteger` to make sure these numbers don't overflow or anything.